### PR TITLE
ref(feedback): fix show/hide logic for trace section, use placeholder while loading

### PR DESCRIPTION
**Pull request by aliu39**
Ref https://github.com/getsentry/sentry/issues/68481

The show/hide logic had a bug where we were showing only if the same-trace issue is a duplicate of a crash report linked error. Which is the opposite of what we want. Example: [JAVASCRIPT-2VE1](https://sentry.sentry.io/feedback/?feedbackSlug=javascript%3A5735318217&project=11276). _Analytics are still correct, but atm no timelines (2+ events) are shown_.

Also: 
- use placeholder when the data is fetching.
- use error boundary.
- remove the `hasProject` prop. It makes the logic hard to read, and was only used for the crash-report-dup condition. Comparing the issue ids is sufficient for this.

### CODE REVIEW

File: static/app/components/feedback/feedbackItem/traceDataSection.tsx [hunk 1]
```diff
@@ -17,23 +18,19 @@ import {useTraceTimelineEvents} from 'sentry/views/issueDetails/traceTimeline/us
...
     event: eventData,
   });
-  const show =
-    !isLoading &&
-    !isError &&
-    traceEvents.length > 1 && // traceEvents include the current event.
-    (!hasProject || !crashReportId || oneOtherIssueEvent?.id === crashReportId);
+  // Note traceEvents includes the current event (feedback).
+
+  const [isCrashReportDup, setIsCrashReportDup] = useState(false);
```
Comment (c298lee) [inline static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: what is this used for?

Comment (aliu39) [reply static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: Whether to show or hide the whole section - see return statement

Comment (aliu39) [reply static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: Oops the condition should be `!isCrashReportDup` !!

Comment (c298lee) [reply static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: what exactly does is crash report dup mean? Sorry I don't really understand the bug

Comment (aliu39) [reply static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: Basically the other section gets error data from a SDK-submitted, explicit event ID, while this section gets it from the trace. In some cases having both would be redundant.

Comment (aliu39) [reply static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: Screenshot 2024-10-17 at 11 23 02 AM
From my linked example

Comment (ryan953) [reply static/app/components/feedback/feedbackItem/traceDataSection.tsx:33]: do we need the state here? it seems like `const isCrashReportDupe = !isError && !isLoading && !!crashReportId && oneOtherIssueEvent?.id === crashReportId;` would be the same thing.

File: static/app/components/feedback/feedbackItem/traceDataSection.tsx [hunk 2]
```diff
@@ -45,23 +43,36 @@ export default function TraceDataSection({
...
+      {isLoading ? (
+        <Placeholder height="114px" />
+      ) : (
+        <IssuesTraceDataSection event={eventData} />
+      )}
     </Section>
   ) : null;
 }
+
+function eventIsCrashReportDup(
```
Comment (ryan953) [inline static/app/components/feedback/feedbackItem/traceDataSection.tsx:73]:
```suggestion
function eventIsCrashReportDupe(
```

**Review by ryan953 — APPROVED** [review]

**Pull request merged**